src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +842,348 @@ def __try_add_error(error_list, detail):
+            return package_list, [], capacity - self.__calc_package_payload_size_on_disk(package_list)
+        if self.__calc_package_payload_size_on_disk(package_list[0]) > capacity:
+            return [], package_list, capacity
+
+        while left_index < right_index:
+            mid_index = left_index + int((right_index - left_index) / 2)
+            if self.__calc_package_payload_size_on_disk(package_list[:mid_index]) >= capacity:
+                right_index = mid_index
+            else:
+                left_index = mid_index + 1
+
+        truncated_list = package_list[:left_index - 1]
+        packages_removed_from_list = package_list[left_index - 1:]
+        truncated_list_byte_size = self.__calc_package_payload_size_on_disk(truncated_list)
+        return truncated_list, packages_removed_from_list, capacity - truncated_list_byte_size
+
+    def __removed_older_complete_status_files(self, status_folder):
+        """ Retain 10 latest status complete file and remove other .complete.status files """
+        files_to_removed = []
+        complete_status_files_list = glob.glob(status_folder + '/' + '*.complete.status')   # Glob return empty list if no file matched pattern
+        complete_status_files_list.sort(key=os.path.getmtime, reverse=True)
+
+        if len(complete_status_files_list) <= 10:
+            return
+
+        for file in complete_status_files_list[10:]:
+            try:
+                if os.path.exists(file):
+                    os.remove(file)
+                    files_to_removed.append(file)
+            except Exception as e:
+                self.composite_logger.log_debug("Error deleting complete status file. [File={0} [Exception={1}]]".format(repr(file), repr(e)))
+
+        self.composite_logger.log_debug("Cleaned up older complete status files: {0}".format(files_to_removed))
+
+    def __calc_status_size_on_disk(self, full_status):
+        """ Calculate status file size in bytes on disk """
+        return len(json.dumps(full_status).encode("utf-8"))
+
+    def __calc_package_payload_size_on_disk(self, package_list):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def size_of_constant_status_data(self, complete_status_file_payload, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without package data  """
+        status_file_no_list_data = complete_status_file_payload
+        if assessment_status_index is not None:
+            assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(status_file_no_list_data, assessment_status_index)
+            assessment_msg_empty_list = self.__update_substatus_msg_packages_and_errors(assessment_substatus_msg_copy, [], assessment_substatus_msg_copy['errors'])
```

Comment:
What is the need to update errors in the substatus? Since the source (i.e. assessment_substatus_msg_copy['errors']) is the same (i.e. assessment_substatus_msg_copy), errors should already be a part of it